src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,303 @@
+import "./BloodPressureVisualization.css";
+import { useEffect, useState } from "react";
+import MyDataHelps from "@careevolution/mydatahelps-js";
+import React from "react";
+import { Button, DateRangeNavigator, LoadingIndicator, Title } from "../../presentational";
+import DumbbellChart from "../../presentational/DumbbellChart";
+import { Dumbbell, ClosedInterval, Axis, DataPoint, DumbbellClass } from "../../presentational/DumbbellChart/DumbbellChart";
+import { addDays, format, isEqual, isMonday, previousMonday, startOfDay } from "date-fns";
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+import { BloodPressureDataParameters, BloodPressureDataPoint, BloodPressureDataProvider } from "../../../helpers/blood-pressure-data-providers/query-blood-pressure";
+
+export enum Category {"Low", "Normal", "Elevated", "Stage 1", "Stage 2", "Crisis", "Unknown"};
+export type BloodPressurePreviewState = "WithData" | "NoData" | "Loading" | "Live";
+
+export interface BloodPressureVisualizationProps {
+    previewState? : BloodPressurePreviewState,
+    previewDataProvider : BloodPressureDataProvider,
```

Comment:
Actually wouldn't have a previewDataProvider here but rather just have the component itself determine the data provider based on the previewState.  

Otherwise in the view builder we will have to supply this as well which seems unnecessary

Reply:
Basically I think "previewState" means we do not even invoke a data provider but instead just populate whatever data makes sense to render that previewState